pypesto/visualize/clust_color.py

```diff
@@ -68,44 +69,137 @@ def assign_clustered_colors(vals):
         return []
 
     # assign clusters
-    clust, clustsize, ind_clust = assign_clusters(vals)
-
-    # assign colors
-    vmax = max(clust) - sum(clustsize == 1)
-    cnorm = colors.Normalize(vmin=0, vmax=vmax)
-    scalarmap = cm.ScalarMappable(norm=cnorm)
-
-    # colors for each cluster and one-size clusters
-    cols_clust = scalarmap.to_rgba(range(vmax + 1))
-
-    # grey color for 1-size clusters as the last color in 'cols_clust'
-    cols_clust[vmax] = (0.7, 0.7, 0.7, 1)
-
-    # pre-array of indices for colors
-    ind_col = np.zeros(len(ind_clust))
-
-    # number of colors assigned to cluster
-    sum_col = 0
-
-    # assign color indices for each point
-    for iind, value_ind in enumerate(ind_clust):
-        # if cluster size > 1
-        if clustsize[value_ind] > 1:
-            # assign a color with is not grey
-            ind_col[iind] = sum_col
-            # if element is not the last one in 'ind_col'
-            if iind < len(ind_clust) - 1:
-                # if the next element does not belongs to the seem cluster
-                if value_ind != ind_clust[iind + 1]:
-                    # use the next color
-                    sum_col = sum_col + 1
-        # if cluster size = 1
+    clust, cluster_size, cluster_indices = assign_clusters(vals)
+
+    # create list of colors, which has the correct shape
+    n_clusters = max(clust) - sum(cluster_size == 1)
+
+    # fill color array from colormap
+    colormap = cm.ScalarMappable().to_rgba
+    color_list = colormap(np.linspace(0., 1., n_clusters))
+
+    # We have clustered the results. However, clusters may have size 1,
+    # so we need to rearrange ind_clust in order to map all 1-element-clusters
+    # to the cluster with index n_clusters
+
+    # create a dummy variable with the colors
+    colors = np.zeros((clust.size, 4))
+
+    # run through the cluster_indices and collect which element
+    # is in 1-element-clusters, and which is in real clusters
+    no_clusters = []
+    real_clusters = []
+    for index in cluster_indices:
+        if cluster_size[index] == 1:
+            no_clusters.append(index)
         else:
-            # use grey
-            ind_col[iind] = vmax
+            real_clusters.append(index)
+    real_clusters = np.unique(real_clusters)
+
+    # assign colors to real clusters
+    for icol, iclust in enumerate(real_clusters):
+        ind_of_iclust = np.argwhere(cluster_indices == iclust).flatten()
+        colors[ind_of_iclust, :] = color_list[icol, :]
+
+    # assing color to non-clustered indices
+    for noclust in no_clusters:
+        ind_of_noclust = np.argwhere(noclust in no_clusters).flatten()
+        colors[ind_of_noclust, :] = [0.7, 0.7, 0.7, 1]
```

Comment:
color to argument with this default?

Reply:
well, surely one could do it...
But that's the color for the multi-starts which don't fall into clusters, hence the probably not properly converged starts... I don't see a reason to color them, or can not think of an application for this option, to be honest...

Reply:
For any other plotting function color palettes can be changes. Only this color cannot.

Reply:
Just wanted to implement it, now I'm realizing:
Doesn't actually work: This color will anyway only be used, if the user doesn't specify a coloring scheme himself...